src/instruction/instructions/store.ts

```diff
@@ -0,0 +1,302 @@
+    opcode = setBits(opcode, this.rnPattern, createLowRegisterBits(removeBracketsFromRegisterString(options[1])))
+    opcode = setBits(opcode, this.immPattern, createImmediateBits(removeBracketsFromRegisterString(options[2]),5))
+    return opcode
+  }
+
+  public executeInstruction(
+      opcode: Halfword,
+      registers: Registers,
+      memory: IMemory
+  ): void {
+    memory.writeWord(
+      registers.readRegister(getBits(opcode,this.rnPattern).value).add(
+          Word.fromUnsignedInteger(getBits(opcode,this.immPattern).value)
+      ),
+      registers.readRegister(getBits(opcode,this.rtPattern).value)
+    )
+  }
+}
+
+
+/**
+ * Represents a 'STORE' instruction - STR (register offset) - word
+ */
+export class StoreInstructionRegisterOffset extends BaseInstruction {
+  public name: string = 'STR'
+  public pattern: string = '0101000XXXXXXXXX'
+  private rnPattern: string = '0101000000XXX000'
+  private rmPattern: string = '0101000XXX000000'
+  private rtPattern: string = '0101000000000XXX'
+  private expectedOptionCount: number = 3
+
+  public canEncodeInstruction(commandName: string, options: string[]): boolean {
+    if (super.canEncodeInstruction(commandName, options)) {
+      if (
+          isOptionCountValid(options, this.expectedOptionCount) &&
+          !isImmediate(options[2]) &&
+          registerStringHasBrackets(options[1],options[2])
+      ) {
+        return true
+      }
+    }
+    return false
+  }
+
+  public encodeInstruction(options: string[], labels: ILabelOffsets): Halfword {
+    checkOptionCount(options,3)
+    let opcode: Halfword = create(this.pattern)
+    opcode = setBits(opcode, this.rtPattern, createLowRegisterBits(removeBracketsFromRegisterString(options[0])))
+    opcode = setBits(opcode, this.rnPattern, createLowRegisterBits(removeBracketsFromRegisterString(options[1])))
+    opcode = setBits(opcode, this.rmPattern, createLowRegisterBits(removeBracketsFromRegisterString(options[2])))
+    return opcode
+  }
+
+  public executeInstruction(
+    opcode: Halfword,
+    registers: Registers,
+    memory: IMemory
+  ): void {
+    memory.writeWord(
+      registers.readRegister(getBits(opcode,this.rnPattern).value).add(
+        registers.readRegister(getBits(opcode,this.rmPattern).value)
+      ),
+      registers.readRegister(getBits(opcode,this.rtPattern).value)
+    )
+  }
+}
+
+/**
+ * Represents a 'STORE' instruction - STR (immediate offset) - halfword
+ */
+export class StoreInstructionImmediateOffsetHalfword extends BaseInstruction {
+  public name: string = 'STRH'
+  public pattern: string = '10000XXXXXXXXXXX'
+  private rnPattern: string = '1000000000XXX000'
+  private rtPattern: string = '1000000000000XXX'
+  private immPattern: string = '10000XXXXX000000'
+  private expectedOptionCount: number = 3
+
+  public canEncodeInstruction(commandName: string, options: string[]): boolean {
+    if (super.canEncodeInstruction(commandName, options)) {
```

Comment:
with artihmetic operation branch just name is used instead of commandName. To keep it uniform change to just name as well.

Or can be done one a general refactor story